File: Longest Palindromic Substring.cpp
class Solution(object):
	def longestPalindrome(self, s):
		"""
		:type s: str
		:rtype: str
		"""
		longestPalindrome = ""
		for i in range(len(s)):
			L = i-1
			R = i+1
			while L >= 0 and R < len(s):
				if s[L] != s[R]:
					break
				L -= 1
				R += 1
			if R-L-1 > len(longestPalindrome):
				longestPalindrome = s[L+1:R]
		for i in range(len(s)):
			L = i-1
			R = i+2
			if i+1 < len(s) and s[i] == s[i+1]:
				while L >= 0 and R < len(s):
					if s[L] != s[R]:
						break
					L -= 1
					R += 1
				if R-L-1 > len(longestPalindrome):
					longestPalindrome = s[L+1:R]
		return longestPalindrome

print Solution().longestPalindrome("cbbd")

File: Longest Palindromic Substring.py
import string


class Solution(object):
	def longestPalindrome(self, s):
		"""
		:type s: str
		:rtype: str
		"""
		longestPalindrome = s[0]
		for i in range(len(s)):
			sPrime = s[i:]
			lastFoundIndex = string.rfind(sPrime, s[i])
			while lastFoundIndex != -1:
				# palindrome is found
				if self.isPalindrome(sPrime[:lastFoundIndex+1]):
					palindrome = sPrime[:lastFoundIndex+1]
					# if found palindrome is longer, update longest
					if len(palindrome) > len(longestPalindrome):
						longestPalindrome = palindrome
					break
				else:
					lastFoundIndex = string.rfind(sPrime, s[i])


	def isPalindrome(self, s):
		if len(s) == 0 or len(s) == 1:
			return True
		if s[0] == s[-1]:
			return isPalindrome(s[1:-1])
		else:
			return False


Solution().longestPalindrome("helloworld")
